src/levels/LevelLoader.java

```diff
@@ -1,266 +1,338 @@
+    
+		LevelFactory f = getLevelFactory()
+										.setName("Sample")
+										.setGridSize(8,8)
+										.setInitResources(200)
+										.setResPerTurn(25)
+										.addAllAllowedPlants(samplePlants)
+										.addAllZombies(sampleZombies);
+		levels.add(f.toLevelInfo());
+		f.toXML();
+		LOG.debug("Added Sample Level");
+	}
+	
+	/**
+	 * Deserializes all LevelInfo from SER files in the levels directory into the game.
+	 */
+	private static void deserializeLevels () {
+		try {
+			JAXBContext jc = JAXBContext.newInstance(LevelInfo.class); //uses JaxB
+			Unmarshaller unM = jc.createUnmarshaller();
+			
+			File folder = new File("levels/");
+			File[] listOfFiles = folder.listFiles();
+
+			if (listOfFiles == null) {
+				throw new IOException("Missing Level Directory");
+			} else if (listOfFiles.length == 0) {
+				throw new IOException("Missing Level Files");
+			} else {
+				for (int i = 0; i < listOfFiles.length; i++) {
+					if (listOfFiles[i].isFile()) {
+						LOG.debug("Attempting to Deserialize LevelInfo from " + listOfFiles[i].getName());
+
+						LevelInfo lvl = (LevelInfo) unM.unmarshal(listOfFiles[i]);
+						levels.add(lvl);
+					}
+				}
+				LOG.debug("Finished Deserialization");
+			}
+		} catch (IOException io) {
+			LOG.error("Failed to deserialize levels - IO Error");
+			io.printStackTrace();
+		} catch (JAXBException e) {
+			LOG.error("Failed to deserialize levels - JaxB Error");
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Must be called before this class is used
+	 */
+	public static void init() {
+		levels = new ArrayList<>();
+		currentLevel = 0;
+
+		//DeSerialize all Level Info into Level Info
+		deserializeLevels();
+		//Generate a Sample level if no serialized level was found
+		if(levels.size()==0) {
+			sampleLevels();
+		}
+	}
+	
+	public static void refreshLevelLists() {
+		levels.clear();
+		init();
+	}
+	
+	/**
+	 * Fetches the LevelInfo for the next Level
+	 * @return LevelInfo. Null if all levels have been played
+	 */
+	public static LevelInfo getNextLevel() {
+		currentLevel++;
+		if (currentLevel < levels.size()) {
+			return levels.get(currentLevel);
+		}
+		return null;
+	}
+	
+	/**
+	 * Fetches the LevelInfo for a particular level
+	 * @param level
+	 * @return
+	 */
+	public static LevelInfo getLevel(int level) {
+		if(level <= getNumLevels()) {
+			currentLevel = level - 1;
+			return levels.get(currentLevel);
+		} else {
+			return null;
+		}
+	}
+	
+	/**
+	 * Returns the number of levels
+	 * @return
+	 */
+	public static int getNumLevels() {
+		return levels.size();
+	}
+	
+	/**
+	 * Returns the number of the current Level
+	 * @return
+	 */
+	public static int getCurrentLevel() {
+		return currentLevel + 1;
+	}
+	
+	/**
+	 * Resets progress through the levels.
+	 */
+	public static void reset() {
+		currentLevel = 0;
+	}
+	
+	/**
+	 * Factory Class for building LevelInfos. Contains some default values.
+	 * @author david
+	 *
+	 */
+	public class LevelFactory {

```

Comment:
By the way, this chained method calls from this class is cool.

Reply:
I don't like factories that spread themselves across multiple lines